src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/RoleAssignment.java

```diff
@@ -22,4 +23,17 @@
     private Instant created;
     private List<String> authorisations;
     private RoleAssignmentAttributes attributes;
+
+    /**
+     * This is how the de-serialization works:
+     * Missing caseId -> caseId == null
+     * caseId: null -> Optional.empty()
+     * caseId: "null" -> Optional.empty()
+     * caseId: "12345" -> Optional.of("12345")
+     */
+    public boolean isCaseRoleAssignment() {
+        return this.getAttributes() != null
+            && this.getAttributes().getCaseId() != null
+            && StringUtils.isNotBlank(this.getAttributes().getCaseId().orElse(""));
```

Comment:
would this work:
```suggestion
            && !this.getAttributes().getCaseId().isEmpty())
```

Reply:
It is not matching the receiving "". This is used to pass an empty string to isNotBlank() method when caseId deserialises to Optional.empty(), meaning no caseId is defined. The expression has to resolve to boolean false.

Reply:
It is actually a shortcut proposed by Kiran instead of two lines:
`&& this.getAttributes().getCaseId().isPresent()
  && StringUtils.isNotBlank(this.getAttributes().getCaseId().get());`

Reply:
I don't think we'll ever receive this:
`caseId : ""`
if a caseId has no value, the we'll get:
`caseId : null`